drivers/hmis/app/models/hmis/reminders/reminder_generator.rb

```diff
@@ -0,0 +1,213 @@
+          .order(AssessmentDate: :desc)
+          .group_by(&:enrollment_id)
+
+        assessments = @all_assessments[enrollment.EnrollmentID]
+        return unless assessments
+
+        stages = data_stages(stages)
+        found = assessments.detect do |r|
+          wip.include?(r.wip) && stages.include?(r.data_collection_stage)
+        end
+        found&.AssessmentDate
+      end
+
+      # Show reminder if ANY client in the household is missing an Annual Assessment within the
+      # range when the annual is due, and today is on or after the start of that range.
+      def annual_assessment_reminder(enrollment)
+        hoh_entered_on = hoh_entry_date(enrollment)
+        return unless hoh_entered_on
+
+        hoh_entered_on = normalize_yoy_date(hoh_entered_on)
+        window = 30.days
+        # not due for an assessment yet
+        return if today < (hoh_entered_on + (1.year - window))
+
+        hoh_anniversary = hoh_entered_on.change(year: today.year)
+        start_date = hoh_anniversary - window
+        due_date = hoh_anniversary + window
+
+        # a relevant assessment ocurred.
+        # FIXME: maybe we don't include assessments that occur after end_date? This might
+        # encourage people to back-date assessments.
+        last_assessed_on = last_assessment_date(enrollment: enrollment, stages: [:annual_assessment], wip: [false])
+        return if last_assessed_on && last_assessed_on >= start_date
+
+        new_reminder(
+          topic: ANNUAL_ASSESSMENT_TOPIC,
+          due_date: due_date,
+          enrollment: enrollment,
+        )
+      end
+
+      # Show reminder if a client in the household has turned 18 since the entry date, and they
+      # have not had either an update or an annual since turning 18.
+      def aged_into_adulthood_reminder(enrollment)
+        age_of_majority = 18
+        client = enrollment.client
+        return unless client.DOB
+
+        entry_date = enrollment.EntryDate
+        adulthood_birthday = normalize_yoy_date(client.DOB) + age_of_majority.years
+        # the client was already an adult on entry date
+        return if adulthood_birthday <= entry_date
+
+        # client had an assessment after they became and adult
+        last_assessed_on = last_assessment_date(enrollment: enrollment, stages: [:update, :annual_assessment], wip: [false])
+        return if last_assessed_on && last_assessed_on >= adulthood_birthday
+
+        new_reminder(
+          topic: AGED_INTO_ADULTHOOD_TOPIC,
+          due_date: adulthood_birthday,
+          enrollment: enrollment,
+        )
+      end
+
+      # Show reminder if: there are any household members that are missing intake assessments OR
+      # have WIP intake assessments
+      def intake_assessment_reminder(enrollment)
+        # there's no submitted assessment
+        return if last_assessment_date(enrollment: enrollment, stages: [:project_entry], wip: [false])
```

Comment:
right? I'm reading the comment as if its for the line below
```suggestion
        # skip if there is a submitted assessment
        return if last_assessment_date(enrollment: enrollment, stages: [:project_entry], wip: [false])
```